### ISSUE TITLE
Fails on macOS 14.1.1 under Python 3.9 and Python 3.12

### ISSUE THREAD

**Issue by jamiejennings** [manzik/cmdbench#5]
`pip install cmdbench` succeeds.

On a system with Python 3.12, this happens:
```
$ cmdbench
Traceback (most recent call last):
  File "/opt/homebrew/bin/cmdbench", line 5, in <module>
    from cmdbench.cli import benchmark
  File "/opt/homebrew/lib/python3.12/site-packages/cmdbench/cli.py", line 8, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

And with Python 3.12, this happens:
```
$ cmdbench --iterations=1 "ls -l >/tmp/ls-l" 

Benchmarking started..
  0%|                                                                                                    | 0/1 [00:00<?, ?it/s]Process Process-2:
Process Process-3:
Traceback (most recent call last):
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 349, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/psutil/_common.py", line 508, in wrapper
    raise raise_from(err, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 349, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/psutil/_common.py", line 506, in wrapper
    return fun(self)
  File "/usr/local/lib/python3.9/site-packages/psutil/_common.py", line 508, in wrapper
    raise raise_from(err, None)
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 375, in _get_kinfo_proc
    ret = cext.proc_kinfo_oneshot(self.pid)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.9/site-packages/psutil/_common.py", line 506, in wrapper
    return fun(self)
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 375, in _get_kinfo_proc
    ret = cext.proc_kinfo_oneshot(self.pid)
ProcessLookupError: [Errno 3] assume no such process (originated from sysctl(kinfo_proc), len == 0)

During handling of the above exception, another exception occurred:

ProcessLookupError: [Errno 3] assume no such process (originated from sysctl(kinfo_proc), len == 0)
Traceback (most recent call last):

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 355, in _init
    self.create_time()
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 757, in create_time
    self._create_time = self._proc.create_time()
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 349, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 355, in _init
    self.create_time()
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 477, in create_time
    return self._get_kinfo_proc()[kinfo_proc_map['ctime']]
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 757, in create_time
    self._create_time = self._proc.create_time()
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 354, in wrapper
    raise NoSuchProcess(self.pid, self._name)
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 349, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 477, in create_time
    return self._get_kinfo_proc()[kinfo_proc_map['ctime']]
  File "/usr/local/lib/python3.9/site-packages/psutil/_psosx.py", line 354, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil.NoSuchProcess: process no longer exists (pid=6528)
psutil.NoSuchProcess: process no longer exists (pid=6528)

During handling of the above exception, another exception occurred:


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/cmdbench/core.py", line 188, in collect_time_series
    p = psutil.Process(shared_process_dict["target_process_pid"])
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 319, in __init__
    self._init(pid)
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 368, in _init
    raise NoSuchProcess(pid, msg=msg)
  File "/usr/local/lib/python3.9/site-packages/cmdbench/core.py", line 126, in collect_fixed_data
    p = psutil.Process(shared_process_dict["target_process_pid"])
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 319, in __init__
    self._init(pid)
  File "/usr/local/lib/python3.9/site-packages/psutil/__init__.py", line 368, in _init
    raise NoSuchProcess(pid, msg=msg)
psutil.NoSuchProcess: process PID not found (pid=6528)
psutil.NoSuchProcess: process PID not found (pid=6528)
Last runtime:  0.003 seconds: 100%|████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.52s/it]
Benchmarking done.

====> First Iteration <====

Process: 
    Stdout: ''
    Stderr: 'ls: >/tmp/ls-l: No such file or directory\n'
    Runtime: 0.003 second(s)
    Exit code: 1

CPU (seconds): 
    User time: 0
    System time: 0
    Total time: 0

Memory (bytes): 
    Maximum: 0
    Maximum per process: 0

Time series: 
    Sampling milliseconds: array([], dtype=float64)
    CPU (percentages): array([], dtype=float64)
    Memory (bytes): array([], dtype=float64)


Done.
$ 
```

**Comment by manzik** [comment]
Hey Jamie. 

Regarding `cmdbench --iterations=1 "ls -l >/tmp/ls-l"` on python 3.9: I wrote this application anticipating use for long-running applications specifically, which is what my bioinformatics lab supervisor needed at the time. When the command you are benchmarking quits too fast, `cmdbench` does not get enough time to start monitoring the command and its process. Hence the error you see (needs error handling).

In the new update of [cmdbench==0.1.20](https://pypi.org/project/cmdbench/0.1.20/), you just see a warning instead:
<img width="859" alt="image" src="https://github.com/user-attachments/assets/328e9c08-ce78-4fc6-bf65-22cc30d49dc1">

Regarding running the command `cmdbench` on python 3.12: The issue has been resolved also in the new update [cmdbench==0.1.20](https://pypi.org/project/cmdbench/0.1.20/). It had to do with setuptools not supporting some of the functionalities `cmdbench` was using starting from python 3.8.

Please let me know if the issue is resolved, or if any of the problems persist.

**Comment by jamiejennings** [comment]
Thanks!  The fix works for me, and now it makes sense why I saw the error.  I have just now tested on some commands that need more than 10ms to complete, and it works well.
